src/somesy/cli/validators.py

```diff
@@ -0,0 +1,59 @@
+"""Functions to validate and prioritize CLI inputs."""
+import logging
+from pathlib import Path
+
+import typer
+
+from somesy.cli.models import SyncCommandOptions
+
+logger = logging.getLogger("somesy")
+
+SYNC_COMMAND_DEFAULTS = {
+    "input_file": Path(".somesy.toml"),
+    "no_sync_cff": False,
+    "cff_file": Path("CITATION.cff"),
+    "no_sync_pyproject": False,
+    "pyproject_file": Path("pyproject.toml"),
+    "show_info": False,
+    "verbose": False,
+    "debug": False,
+}
+
+
+def get_prioritized_sync_command_inputs(
+    cli_options: SyncCommandOptions, file_options: SyncCommandOptions
+) -> dict:
+    """Prioritize sync command inputs and return prioritized inputs.
+
+    Check direct cli inputs which has the highest priority, then check file inputs and both of them does not have the config, use defaults.
+
+    Args:
+        cli_options (dict): _description_
+        file_options (_type_): _description_
+
+    Returns:
+        dict: _description_
+    """
+    validated_inputs = {}
+
+    for key, value in SYNC_COMMAND_DEFAULTS.items():
+        if getattr(cli_options, key) is not None:
+            validated_inputs[key] = getattr(cli_options, key)
+        elif getattr(file_options, key) is not None:
+            validated_inputs[key] = getattr(file_options, key)
+        else:
+            validated_inputs[key] = value
+
+    # if a output file is given by cli input, use it
+    if cli_options.cff_file is not None:
+        validated_inputs["no_sync_cff"] = False
+    if cli_options.pyproject_file is not None:
+        validated_inputs["no_sync_pyproject"] = False
+
+    # at least one of the sync options must be enabled
+    if validated_inputs["no_sync_cff"] and validated_inputs["no_sync_pyproject"]:
+        logger.warning("There should be at least one file to sync.")
+        typer.Exit(code=0)
+
+    logger.debug(f"validated inputs: {validated_inputs}")
+    return validated_inputs
```

Comment:
Is this method really needed?

You probably should just update your empty/loaded config with all non-default values passed in the CLI, similarly as described here https://github.com/pydantic/pydantic/discussions/3139#discussioncomment-4797649

i.e. turn the cli options model into `dict` with `exclude_defaults=True`, and use `update` on your file options model.

Reply:
So I'd streamline that to:

* having your Options class with the defaults
* updating it as described above from a detected somesy config file
* updating it the same way from cli arguments, if given
* running the validation on the result

Or something like that.

Reply:
I realize it already has a pydantic model so, this dataclass and validation function is completely redundant. I will update it accordingly.

Reply:
@apirogov updated accordingly with [this commit](https://github.com/Materials-Data-Science-and-Informatics/somesy/pull/11/commits/422938d9abe22cc93b94e31e6419a0a87ef3c090)